bitcoin/src/blockdata/transaction.rs

```diff
@@ -833,6 +834,124 @@ impl Transaction {
     pub fn script_pubkey_lens(&self) -> impl Iterator<Item = usize> + '_ {
         self.output.iter().map(|txout| txout.script_pubkey.len())
     }
+
+    // TODO: Find a better reference for how taproot sigops are limited.
+    /// Counts the total number of sigops.
+    ///
+    /// This value is for pre-taproot transactions only.
+    ///
+    /// > In taproot, a different mechanism is used. Instead of having a global per-block limit,
+    /// > there is a per-transaction-input limit, proportional to the size of that input.
+    /// > ref: <https://bitcoin.stackexchange.com/questions/117356/what-is-sigop-signature-operation#117359>
+    ///
+    /// The `spent` parameter is an optional closure/function that takes in an [`OutPoint`] and
+    /// returns a [`TxOut`]. Without access to the previous [`TxOut`], any sigops in redeemScripts,
+    /// witnessScripts, and P2WPKH sigops will not be counted.
+    pub fn total_sigop_cost<S>(&self, mut spent: Option<S>) -> usize
```

Comment:
In b5347f8d754d6c2daa287d838af5dcfde1910cad:

Ok, now that I'm reading the code more closely, you should just drop the `Option` here. If people don't want to proide a function they can write `|_| None`. We could also provide a constant for this if you want.